4/m_ipinfo_io.cpp

```diff
@@ -0,0 +1,213 @@
+    * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    */
+
+    /// $ModAuthor: Jean Chevronnet (reverse) <mike.chevronnet@gmail.com>
+    /// $ModDesc: Ip information from Ipinfo.io in /WHOIS (only irc operators), found more information at https://ipinfo.io/developers.
+    /// $ModDepends: core 4
+    /// $ModConfig: <ipinfo apikey="YOUR IP INFO.IO APIKEY">
+    /// $CompilerFlags: find_compiler_flags("RapidJSON")
+    /// $CompilerFlags: find_compiler_flags("libcurl")
+    /// $LinkerFlags: find_linker_flags("libcurl")
+
+    #include "inspircd.h"
+    #include "extension.h"
+    #include "modules/httpd.h"
+    #include "modules/whois.h"
+    #include "thread.h"
+    #include <rapidjson/document.h>
+    #include <rapidjson/error/en.h>
+    #include <curl/curl.h>
+    #include <mutex>
+    #include <regex>
+    #include <fmt/core.h>
+
+    class IPInfoResolver : public Thread
+    {
+    private:
+        std::mutex mtx;
+        StringExtItem& cachedinfo;
+        std::string apikey;
+        std::string theiruuid;
+        User* resolved_user; // Renamed to avoid shadowing
+
+        static size_t WriteCallback(void* contents, size_t size, size_t nmemb, std::string* s)
+        {
+            s->append(static_cast<char*>(contents), size * nmemb);
+            return size * nmemb;
+        }
+
+        void OnStart() override
+        {
+            CURL* curl;
+            CURLcode res;
+            std::string response;
+
+            curl_global_init(CURL_GLOBAL_DEFAULT);
+            curl = curl_easy_init();
+            if (curl)
+            {
+                std::string url = "https://ipinfo.io/" + resolved_user->client_sa.addr() + "?token=" + apikey;
+                curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+                curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
+                curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response);
+                res = curl_easy_perform(curl);
+                if (res != CURLE_OK)
+                {
+                    std::lock_guard<std::mutex> lock(mtx);
+                    ServerInstance->SNO.WriteGlobalSno('a', fmt::format("IPInfo: Failed to get data for {}: {}", resolved_user->nick, curl_easy_strerror(res)));
+                }
+                else
+                {
+                    ParseResponse(resolved_user, response);
+                }
+                curl_easy_cleanup(curl);
+            }
+            curl_global_cleanup();
+        }
+
+        void ParseResponse(User* user, const std::string& response)
+        {
+            rapidjson::Document document;
+            if (document.Parse(response.c_str()).HasParseError())
+            {
+                std::lock_guard<std::mutex> lock(mtx);
+                ServerInstance->SNO.WriteGlobalSno('a', fmt::format("IPInfo: Failed to parse JSON for {}: {}", user->nick, rapidjson::GetParseError_En(document.GetParseError())));
```

Comment:
The method already takes a fmtlib format string there's no need to call fmt::format here.